src/code42cli/cmds/devices.py

```diff
@@ -0,0 +1,399 @@
+from code42cli.errors import Code42CLIError
+from code42cli.file_readers import read_csv_arg
+from code42cli.options import format_option
+from code42cli.options import sdk_options
+from code42cli.output_formats import OutputFormatter
+
+
+@click.group(cls=OrderedGroup)
+@sdk_options(hidden=True)
+def devices(state):
+    """For managing computers within Code42."""
+    pass
+
+
+device_id_option = click.option(
+    "--device-id",
+    required=True,
+    type=str,
+    help="The device ID for the device on which you wish to act.",
+)
+
+change_device_name_option = click.option(
+    "--change-device-name",
+    required=False,
+    is_flag=True,
+    default=False,
+    help='Include to prepend "deactivated_" and today\'s date to the name of any deactivated devices.',
+)
+
+purge_date_option = click.option(
+    "--purge-date",
+    required=False,
+    type=str,
+    default=None,
+    help="The date on which the archive should be purged from cold storage in yyyy-MM-dd format. If not provided, the date will be set according to the appropriate org settings.",
+)
+
+
+@devices.command()
+@device_id_option
+@change_device_name_option
+@purge_date_option
+@sdk_options()
+def deactivate(state, device_id, change_device_name, purge_date):
+    """Deactivate a device within Code42"""
+    _deactivate_device(state.sdk, device_id, change_device_name, purge_date)
+
+
+def _deactivate_device(sdk, device_id, change_device_name, purge_date):
+    try:
+        sdk.devices.deactivate(device_id)
+    except exceptions.Py42BadRequestError:
+        raise Code42CLIError("The device {} is in legal hold.".format(device_id))
+    except exceptions.Py42NotFoundError:
+        raise Code42CLIError("The device {} was not found.".format(device_id))
+    except exceptions.Py42ForbiddenError:
+        raise Code42CLIError("Unable to deactivate {}.".format(device_id))
+    if purge_date or change_device_name:
+        device = sdk.devices.get_by_id(device_id)
+        guid = device.data["guid"]
+        name = device.data["name"]
+    if purge_date:
+        _update_cold_storage_purge_date(sdk, guid, purge_date)
+    if change_device_name and not name.startswith("deactivated_"):
+        _change_device_name(
+            sdk, guid, "deactivated_" + date.today().strftime("%Y-%m-%d") + "_" + name
+        )
+
+
+def _update_cold_storage_purge_date(sdk, guid, purge_date):
+    archives_response = sdk.archive.get_all_by_device_guid(guid)
+    archive_guid_list = [
+        archive["archiveGuid"]
+        for page in archives_response
+        for archive in page["archives"]
+    ]
+    for archive_guid in archive_guid_list:
+        sdk.archive.update_cold_storage_purge_date(archive_guid, purge_date)
+
+
+def _change_device_name(sdk, guid, name):
+    device_settings = sdk.devices.get_settings(guid)
+    device_settings.name = name
+    sdk.devices.update_settings(device_settings)
+
+
+_DEVICE_INFO_KEYS_MAP = OrderedDict()
```

Comment:
If this is only used in a single function, it would make sense to localize there, either by moving this until the command logic or wrapping it a helper function and calling it within the command

Reply:
Moved it into the function.